### PR TITLE
nrf54l: Removing hardcoded clock frequency

### PR DESCRIPTION
**Pull request by degjorva**
Adding in config that takes clock-frequency from device tree and passes it to TF-m.
Removing hardcoded NRF_CONFIG_CPU_FREQ_MHZ

### CODE REVIEW

File: modules/trusted-firmware-m/tfm_boards/CMakeLists.txt [hunk 1]
```diff
@@ -71,6 +71,7 @@ target_sources(tfm_sprt
     $<$<NOT:$<BOOL:${TFM_SP_LOG_RAW_ENABLED}>>:${CMAKE_CURRENT_SOURCE_DIR}/common/dummy_tfm_sp_log_raw.c>
 )
 
+
```
Comment (tomi-font) [inline modules/trusted-firmware-m/tfm_boards/CMakeLists.txt:74]:
```suggestion
```

File: modules/trusted-firmware-m/Kconfig [hunk 2]
```diff
@@ -291,6 +291,18 @@ config TFM_PROFILE_TYPE_MINIMAL
 	  and reproduce the desired configuration through kconfig fragments.
 endchoice
 
+config TFM_CPU_FREQ_MHZ
+	int
+	default $(div, $(mul, 1, $(dt_node_int_prop_int,/cpus/cpu@0,clock-frequency)), 1000000) \
```
Comment (tomi-font) [inline modules/trusted-firmware-m/Kconfig:296]: Why a multiplication by 1?

File: modules/trusted-firmware-m/Kconfig [hunk 3]
```diff
@@ -291,6 +291,18 @@ config TFM_PROFILE_TYPE_MINIMAL
...
+		if $(dt_node_has_prop,/cpus/cpu@0,clock-frequency)
+	default $(div, $(mul, 1, $(dt_node_int_prop_int,/cpus/cpu@1,clock-frequency)), 1000000) \
+		if $(dt_node_has_prop,/cpus/cpu@1,clock-frequency)
+	default 128
+	help
+	  Gives access to the cpu-clock frequency from the device tree when
+	  building with TF-m.
+
+
```
Comment (tomi-font) [inline modules/trusted-firmware-m/Kconfig:305]:
```suggestion
```

File: modules/trusted-firmware-m/Kconfig (hunk 2 above)
Comment (degjorva) [reply modules/trusted-firmware-m/Kconfig]: Great question. I got a type error when using div on the value from the dt_node, but not with mul. But when I tried it now to get the exact error it worked perfectly fine. So removed it.

File: modules/trusted-firmware-m/Kconfig [hunk 4]
```diff
@@ -291,6 +291,17 @@ config TFM_PROFILE_TYPE_MINIMAL
...
 
+config TFM_CPU_FREQ_MHZ
+	int
+	default $(div, $(dt_node_int_prop_int,/cpus/cpu@0,clock-frequency), 1000000) \
+		if $(dt_node_has_prop,/cpus/cpu@0,clock-frequency)
+	default $(div, $(dt_node_int_prop_int,/cpus/cpu@1,clock-frequency), 1000000) \
+		if $(dt_node_has_prop,/cpus/cpu@1,clock-frequency)
+	default 128
+	help
+	  Gives access to the cpu-clock frequency from the device tree when
```
Comment (tomi-font) [inline modules/trusted-firmware-m/Kconfig:302]:
```suggestion
	  Gives access to the CPU clock frequency from the devicetree when
```

**Review by tomi-font — APPROVED** [review]

**Review by Vge0rge — APPROVED** [review]

**Pull request merged**